llama.cpp

```diff
@@ -8123,20 +8156,25 @@ struct llm_build_context {
                 // skip computing output for unused tokens
                 struct ggml_tensor * inp_out_ids = build_inp_out_ids();
                 cur   = ggml_get_rows(ctx0,   cur, inp_out_ids);
+                inpL  = ggml_get_rows(ctx0,  inpL, inp_out_ids);
                 inpSA = ggml_get_rows(ctx0, inpSA, inp_out_ids);
             }
-
-            struct ggml_tensor * ffn_inp = ggml_add(ctx0, cur, inpSA);
-            cb(ffn_inp, "ffn_inp", il);
+            struct ggml_tensor * attn_out = cur;
+            struct ggml_tensor * ffn_inp  = ggml_add(ctx0, attn_out, inpL);
+            cb(cur, "ffn_inp", il);
```

Comment:
The wrong tensor is passed to this `cb` naming the `ffn_inp` node.

`attn_out` also isn't used anywhere else anymore except right after it's defined, so it could be removed too, since it's probably already clear enough that `cur` is the attention output in this particular place.

```suggestion
            struct ggml_tensor * ffn_inp  = ggml_add(ctx0, cur, inpL);
            cb(ffn_inp, "ffn_inp", il);
```

Reply:
Fixed !

Reply:
Thanks! (but the note regarding spacing hasn't been addressed)

Reply:
Fixed !